### PR TITLE
Memory Cards: Implement conversion in Qt

### PR DESCRIPTION
**Pull request by RedPanda4552**
### Description of Changes
Implements the memory card conversion option in Qt.

### Rationale behind Changes
Memory card conversion was not implemented in the initial move from wx to Qt.

### Suggested Testing Steps
Convert some memory cards, check that games still see their data on the converted cards.

### CODE REVIEW

File: pcsx2-qt/Settings/MemoryCardConvertDialog.h [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+	QString m_selectedCard;
+	AvailableMcdInfo m_srcCardInfo;
+
+	QString m_destCardName;
+	AvailableMcdInfo m_destCardInfo;
+
+	MemoryCardType m_type = MemoryCardType::File;
+	MemoryCardFileType m_fileType = MemoryCardFileType::PS2_8MB;
+};
```
Comment (F0bes) [inline pcsx2-qt/Settings/MemoryCardConvertDialog.h:52]: Missing EOF newline

**Comment by JordanTheToaster** [comment]
Doesn't appear to work tried to convert a 32MB card to a folder and Burnout 3 saw it as unformatted.

**Review by JordanTheToaster — APPROVED** [review]
All good here now LGTM.

**Review by Mrlinkwii — APPROVED** [review]

**Review by lightningterror — COMMENTED** [review]
Commit can be named properly with prefix.

**Review by refractionpcsx2 — CHANGES_REQUESTED** [review]
Gave it a quick test, and I don't think the conversion from folder to memcard is as good as it could be.

At the moment, if you have too much in your folder, it spams the console (which the user might not even have open) then it says "conversion complete!" and the user might delete the folder thinking everything is peachy, especially if they don't have the console window open.

What I would like to see:
1. When converting to memcard, check if there's enough space for the current game on the card, if not, start a new card, so like `folder_converted_1.mcd` or something.  
2. Maybe make the message to the user a bit more verbose, like "Due to the size of the folder, multiple memcards had to be created" or something.  I dunno, creative freedom :P

**Comment by RedPanda4552** [comment]
@refractionpcsx2 

- Picklist now filters options based on computed space requirements of the card
- Fixed an issue where a card containing more than 512 individual files cannot be opened (by basically changing the default number of files Windows allows a C family program to open from 512 to 1024, equal to that of Unix platforms)
- Added some notes to the UI about it stalling while processing the conversion
- Conversion menu will now refuse to open if folder memcard exceeds 64 MB of data, since no file types can hold more than that atm

**Comment by JordanTheToaster** [comment]
Converted a few of my memory cards including a 32MB card and it seems to read and save fine looks good to me.

**Comment by Mrlinkwii** [comment]
Converted a few of my memory cards including a 64MB memcard  , games seem happy

**Review by lightningterror — COMMENTED** [review]
Commit names should still be properly named, maybe squash some too.

**Comment by RedPanda4552** [comment]
Squashed.

**Review by lightningterror — COMMENTED** [review]
`Qt: Add dialog for memory card conversion.`

Will be better for consistency.

**Review by lightningterror — APPROVED** [review]

**Comment by kenshen112** [comment]
Shoved my ABSOULTE UNIT cards into the QT converter. Flipped to folder and flipped back to image with no seeming issues. Was able to browse in BIOS and see sizes and order of saves remained the same

**Comment by refractionpcsx2** [comment]
LGTM, nice work.

If you could squash things down and sort out your commit names, we can go ahead and merge :)

**Review by refractionpcsx2 — APPROVED** [review]

**Comment by RedPanda4552** [comment]
Squashed and rebased

**Pull request merged**